### ISSUE TITLE
Python: grpclb does not work after upgrade to 1.21.1

### ISSUE THREAD

**Issue by blazej24** [grpc/grpc#19401]
### What version of gRPC and what language are you using?
 Python, 1.16.0 - works
 Python, 1.21.1 - does not work
 
### What operating system (Linux, Windows,...) and version?
 Linux Ubuntu 18.04
 
### What runtime / compiler are you using (e.g. python version or version of gcc)
 Python 3.6.7
 
### What did you do?
I have simply enabled ares resolver through an environment variable: GRPC_DNS_RESOLVER=ares, and then connected to an example service: dns:///hello.mimgrpc.me
 
### What did you expect to see?
I expect that the library will resolve my service name to a list of load balancers, and then connect to my backend server. It works fine with grpcio == 1.16.0
 
### What did you see instead?
On grpcio == 1.21.1 I get 
`error: <_Rendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "DNS resolution failed"
	debug_error_string = "{"created":"@1560946416.044804550","description":"Failed to pick subchannel","file":"src/core/ext/filters/client_channel/client_channel.cc","file_line":3381,"referenced_errors":[{"created":"@1560946416.044801776","description":"Resolver transient failure","file":"src/core/ext/filters/client_channel/resolving_lb_policy.cc","file_line":326,"referenced_errors":[{"created":"@1560946416.044801464","description":"DNS resolution failed","file":"src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc","file_line":358,"grpc_status":14,"referenced_errors":[{"created":"@1560946416.044793671","description":"C-ares status is not ARES_SUCCESS: DNS server returned answer with no data","file":"src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc","file_line":244}]}]}]}"`
 
### Anything else we should know about your project / environment?
Nope.

**Comment by lidizheng** [comment]
@blazej24 Based on the log, I see `DNS server returned answer with no data`. Can you give more detail about how you setup the DNS server?

Also, are you using the same code to against `v1.16.0` and `v1.21.1`? 

If possible, can you turn on the trace log and run your tests again? To turn on trace, set two environmental variable `GRPC_VERBOSITY=DEBUG` and `GRPC_TRACE=all`.

CC @AspirinSJL

**Comment by gnossen** [comment]
@blazej24 Does your setup work with the native resolver?

CC @apolcyn

**Comment by AspirinSJL** [comment]
Looks like a resolver issue.

**Comment by apolcyn** [comment]
TLDR; since the 1.16 branch, two important things in resolver changed:
* c-ares was made the default resolver (on most platforms - doesn't include iOS, Android, g-event or Node)
* SRV queries were made to be opt-in (breaking change to c-ares resolver behavior; done as a part of setting is as default).

See https://github.com/grpc/grpc/pull/17723 and it's description: #17723 disabled SRV queries on the c-ares resolver by default and made them opt in (the native resolver has/will never perform SRV queries). So, in general, clients that want to use grpclb with DNS will need to set the `grpc.dns_enable_srv_queries` channel arg on their channels.

Also note that c-ares was made the default resolver since the 1.16 branch, so setting `GRPC_DNS_RESOLVER=ares` is now basically a no-op

**Comment by blazej24** [comment]
Ok, it works, thanks for the info.

**Comment by kr-ish** [comment]
I don't think I follow entirely- @blazej24 what did you do to get this to work with 1.21.1? Thanks
